src/ripple/protocol/impl/STObject.cpp

```diff
@@ -630,6 +645,13 @@ STObject::getFieldArray(SField const& field) const
 void
 STObject::set(std::unique_ptr<STBase> v)
 {
+    set(v.get());
+}
+
+void
+STObject::set(STBase* v)
```

Comment:
Why does this take a pointer as an argument? It seems like `STBase&&` would be clearer.

Reply:
This function was already declared in `STObject` but it was not defined. I just added the implementation.

Reply:
I'm still not understanding why this interface uses a pointer and not an rvalue ref. rvalue ref seems better given how this is implemented. Am I missing something?

Reply:
Changed to rvalue ref.